src/client/telemetry/importTracker.ts

```diff
@@ -164,12 +143,12 @@ export class ImportTracker implements IExtensionSingleActivationService, INotebo
         // Make sure this is a Python file.
         if (path.extname(document.fileName) === '.py') {
             this.scheduleDocument(document);
-        }
-    }
-
-    private onOpenedOrClosedNotebook(e: INotebookEditor) {
-        if (e.file) {
-            this.scheduleCheck(e.file.fsPath, this.checkNotebook.bind(this, e));
+        } else if (
+            document.notebook &&
+            isJupyterNotebook(document.notebook) &&
+            document.languageId === PYTHON_LANGUAGE
+        ) {
+            this.scheduleDocument(document);
```

Comment:
I assume this is the fix for package import telemetry not showing up?

Reply:
**This fix is for when running cells**, we're already sending telemetry for Opening & Closing notebook.s
I think they may be checking telemetry in Python repo (coz if they're talking to Luciana and looking at their telelemtry, well then it comes from Python repo).

& Possible they haven't looked at our repo for the telemtry (explains why they do'nt see telemetry for open/close of notebooks)

Reply:
Oh and I wanted to ensure we only look for Jupyter notebooks with Python cells, previously it was any notebook.